mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/search/ElasticSearchSearcher.java

```diff
@@ -96,18 +97,24 @@ private SearchRequest prepareSearch(Collection<MailboxId> mailboxIds, SearchQuer
             .map(SortConverter::convertSort)
             .forEach(searchSourceBuilder::sort);
 
-        return new SearchRequest(aliasName.getValue())
+        SearchRequest request = new SearchRequest(aliasName.getValue())
             .types(NodeMappingFactory.DEFAULT_MAPPING_NAME)
             .scroll(TIMEOUT)
-            .source(searchSourceBuilder)
-            .routing(toRoutingKeys(mailboxIds));
+            .source(searchSourceBuilder);
+
+        return toRoutingKey(mailboxIds)
+            .map(request::routing)
+            .orElse(request);
     }
 
-    private String[] toRoutingKeys(Collection<MailboxId> mailboxIds) {
-        return mailboxIds.stream()
-            .map(routingKeyFactory::from)
-            .map(RoutingKey::asString)
-            .toArray(String[]::new);
+    private Optional<String[]> toRoutingKey(Collection<MailboxId> mailboxIds) {
+        if (mailboxIds.size() < MAX_ROUTING_KEY) {
+            return Optional.of(mailboxIds.stream()
+                .map(routingKeyFactory::from)
+                .map(RoutingKey::asString)
+                .toArray(String[]::new));
+        }
+        return Optional.empty();
```

Comment:
does the search result of the same query can be different if I don't specify any routing?

Reply:
No, as no routing means "all shards" including the one owning your data.